### DISCUSSION TITLE
Left Hand IK on gun grips / two handed weapons?

### DISCUSSION THREAD

**Discussion by xkamronx** [Sixze/ALS-Refactored#332] [Q&A]
I've been struggleboating with this for a few months now. Has anyone set up left hand IK stuff for holding two handed objects based on a socket location on another actor successfully? None of the tutorials for ALS v4 Community seem to work with the changes made to Refactored.

**Comment by ameaninglessname** [comment]
I haven't done this myself from scratch, but I think the general idea is IK the right hand to the offset of right hand socket to the root of the item, IK the left hand to the offset of left hand socket to right hand socket. (the item is attached to the hand socket of character mesh)

The fist step could be omitted if the right hand socket is the item root where only left hand IK is needed.

**Reply by xkamronx** [reply]
I've been able to actually move the hand to the socket on the attachment / gun, but it just doesn't seem to work well. Positioning it on the grip while aiming, but it offsets while not in aiming stance and the hand comes off the socket for some reason.

Also still not positive how to correctly replicate the IK for multiplayer.

I've read that using the virtual bones is supposed to make it smoother, but just can't figure out the correct method + bones to use with Refactored to make it work as I want it to.
